src/main/java/janggi/Board.java

```diff
@@ -0,0 +1,74 @@
+package janggi;
+
+import janggi.coordinate.Position;
+import janggi.piece.Piece;
+import janggi.piece.Pieces;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<Position, Piece> positionToPiece;
+
+    private Board(final Map<Position, Piece> positionToPiece) {
+        this.positionToPiece = positionToPiece;
+    }
+
+    public static Board from(final Pieces pieces) {
+        HashMap<Position, Piece> positionToPiece = new HashMap<>();
+
+        pieces.getPieces().forEach(piece ->
+                positionToPiece.put(piece.getPosition(), piece));
+
+        return new Board(positionToPiece);
+    }
```

Comment:
https://github.com/woowacourse/java-janggi/pull/67#discussion_r2007610275

추가 답변 남겨봅니다. 생성자는 객체의 초기화만 진행하여 최대한 빠르게 동작해야한다고 생각하는 분들도 있지만, 개인적으로는 그러한 생성자를 만들기 위해서 정적 팩터리 메서드를 추가하는건 과하지 않나라고 생각합니다.

아래의 코드를 보고 간단하게 생성자로 작성할 수 있는데, 굳이 정적 팩터리 메서드로 Map을 변환하는 과정을 분리해야할 지 고민해보면 좋을 것 같습니다. 

```suggestion
    public Board(final Pieces pieces) {
        this.positionToPiece = pieces.getPieces()
                .stream()
                .collect(
                        Collectors.toMap(
                                Piece::getPosition,
                                piece -> piece
                        )
                );
    }
```

> 하지만 현재 대부분의 클래스가 정적 팩터리 메서드를 사용하고 있고, 그 흐름 안에서 이 클래스만 new를 사용하는 건 일관성 측면에서 오히려 눈에 띄고 다르게 느껴질 수 있습니다.

정말 대부분의 클래스가 정적 팩터리 메서드가 필요한지도 같이 고민해보면 좋을 것 같아요. 불필요한 정적 팩터리 메서드인데, 일관성 측면만 고려하여 나오게된 결과물이 아닌지도 생각해보면 좋을 것 같아요!

남겨 놓은 피드백은 완벽한 정답을 찾는 과정이 아니고 스스로 스타일을 찾아가는 시간이라고 생각하면 좋을 것 같아요~

Reply:
감사합니다! 사실 정적 팩터리 메서드를 활용할 명확한 사례를 보고 판단하면 쉬운 일이지만, 실제로 코드를 작성할 때 항상 엄격하게 고민할 만큼 중요한 문제인지 생각해 봤을 때, 그렇지 않다고 판단했기에 고민보다는 일관성을 지키는 쪽을 택한 것 같습니다.

그리고 말씀드리지 않았던 저의 기준이 하나 더 있습니다. `필드의 타입과 생성자의 파라미터 타입을 일치시키는 것입니다.` 그렇기에 필드가 Map, 전달받는 인자가 List라면 정적 팩토리 메서드를 활용했던 것 같습니다.

Reply:
정적 팩터리 메서드는 충분히 얘기를 나눈 것 같네요.
추가로 아래의 내용만 더 얘기를 나눠보고 싶네요.

> 필드의 타입과 생성자의 파라미터 타입을 일치시키는 것입니다.

필드 타입과 생성자의 파라미터 타입을 일치시키게 되면 외부에서 객체의 내부 자료구조를 다 알게 되는거 아닌가요??

Reply:
> 필드 타입과 생성자의 파라미터 타입을 일치시키게 되면 외부에서 객체의 내부 자료구조를 다 알게 되는거 아닌가요??

맞습니다! 생성자의 파라미터 타입을 일치시키면, 외부에서 객체의 내부 자료구조를 모두 알게 됩니다. 간단한 값 객체가 아닌 경우에 이런 상황을 피하고 싶었습니다. 그렇기에 정적 팩토리 메서드를 사용했습니다.

생성자의 파라미터 타입을 일치시키지 않는다면 아래 예시와 같이 상속을 받은 경우, 생성자의 바디를 읽어야지만 어떤 필드를 가지는지 확인할 수 있다고 생각했습니다. 그렇기에 생성자의 파라미터 타입을 일치시키면 필드를 명확히 확인할 수 없는 상태(상속)에서, 생성자의 파라미터가 필드의 값과 같음을 암시할 수 있습니다.

또, 일관성을 위해서 정적 메서드를 사용하자는 것을 컨벤션으로 약속한다면, new 키워드가 나올 때 이것이 특별한 객체거나, 외부 라이브러리임을 암시한다고 생각했습니다. 물론 이것은 많은 경험에서 나온 기준이 아니기에, 잘못된 기준일 수 있다고 생각합니다.

```java
public class Cannon extends Piece {

    // 상위 클래스나 바디를 확인하지 않아도 나는 파라미터가 필드와 일치한다고 믿을 수 있다.
    private Cannon(final Position position,
                   final Team team,
                   final MoveStrategy moveStrategy,
                   final RequiredBlockCountStrategy blockStrategy) {
        super(position, team, moveStrategy, blockStrategy);
    }
    ...
}
```

```java
public class Cannon extends Piece {

    // 바디를 확인해서 필드를 파악할 수 있다.
    public Cannon(final Position position,
                   final Team team) {
        super(position, team, new StraightMoveStrategy(), new RequiredBlockCountStrategy());
    }
    ...
}
```

Reply:
> 맞습니다! 생성자의 파라미터 타입을 일치시키면, 외부에서 객체의 내부 자료구조를 모두 알게 됩니다. 간단한 값 객체가 아닌 경우에 이런 상황을 피하고 싶었습니다. 그렇기에 정적 팩토리 메서드를 사용했습니다.

정적 팩터리 메서드를 사용한 이유를 이제야 이해했네요. 파라미터를 동일하게 유지하여 상속을 사용했을 때 필드를 파악하기 쉬울걸로 생각하여 파라미터를 맞췄다는 것도 이해했습니다.

가독성을 위해 좋은 시도일 수는 있으나, 한가지 의견을 하나 드리면 강산이 생각한 컨벤션을 유지하는 것은 어려울 수도 있습니다. 컴파일 에러도 나오지 않으며, 코드 리뷰와 같은 시간에 누군가 눈으로 일일이 확인하여야 컨벤션을 유지할 수 있기 때문입니다. 또한, 처음 누군가가 코드를 살펴본다면 하위 타입에서 super로 초기화하는 값들이 부모 타입의 필드임을 쉽게 알지 못할 수도 있습니다. 결국 계속해서 코드를 살피다가 어느 순간 같은 필드임을 깨닫게 되겠죠. 하지만, 그 타이밍에 누군가가 컨벤션을 놓치는 순간이 발생하면 다시 그러한 약속은 믿지 못하게 되어 부모 타입을 함께 살펴보게 될 수도 있습니다.

강산의 생각이 잘못된 건 아닙니다. 정말 필요한 컨벤션이라면 팀원 모두가 지킬 수 있으니깐요. 하지만, 누군가 반대되는 생각을 가지고 놓치기 시작한다면 적용하기 어려운 컨벤션이 될 수도 있습니다. 이번 미션에서는 페어와 같은 팀을 이뤄 진행했으니, 설명해주신 컨벤션을 서로 지키는 걸로 생각하겠습니다!

Reply:
베루스의 관점에 이어 실무적인 요소까지 짚어주셔서 감사합니다. 말씀해주신 내용 대부분에 공감합니다.
저 역시 정적 팩터리 메서드를 무조건 선호한다기보다는, 사용하다 보니 자연스럽게 익숙해진 정도였고, 만약 팀의 컨벤션이나 현실적인 여건상 맞추기 어렵다면 그에 맞춰 유연하게 대응할 수 있을 것 같습니다.

앞으로 비슷한 상황에서 베루스의 피드백이 떠오를 것 같네요. 감사합니다 ☺️